### PR TITLE
[jobs] Add headers field to JobSubmissionClient and apply to all requests

### LINKED ISSUE #20595: Authentication with remote ray cluster for job submission
We need to provide ability to pass in header for each request made to job server.

### PR DESCRIPTION
**Pull request by jiaodong**
## Why are these changes needed?

This is needed in case user needs to connect with their internal cluster that requires authentication.

Closes #20595 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — COMMENTED** [review]
As discussed offline, let's unify this with the existing `parse_cluster_info` and have @nikitavemuri review!

**Comment by jiaodong** [comment]
cc: @nikitavemuri do you mind taking a look and see if this looks good for CLI for both surfaces ?

File: dashboard/modules/job/sdk.py [hunk 1]
```diff
@@ -27,10 +27,17 @@ class ClusterInfo:
     address: str
     cookies: Optional[Dict[str, Any]]
     metadata: Optional[Dict[str, Any]]
+    headers: Optional[Dict[str, Any]]
```
Comment (nikitavemuri) [inline dashboard/modules/job/sdk.py:30]: Let's also give all additional fields a default value, so this still works if some modules are returning an older version of `ClusterInfo`.

**Review by nikitavemuri — APPROVED** [review]
lgtm!

**Review by edoakes — APPROVED** [review]

**Pull request merged**